tests/core/test_lightning_module.py

```diff
@@ -299,3 +303,39 @@ def assert_device(device: torch.device) -> None:
     assert_device(torch.device("cpu"))
     trainer.predict(model, dataloaders=model.train_dataloader())
     assert_device(torch.device("cpu"))
+
+
+class BoringModelWithShardedTensor(BoringModel):
+    def __init__(self, spec):
+        super().__init__()
+        self.sharded_tensor = dist._sharded_tensor.empty(spec, 10, 20)
+        self.sharded_tensor.local_shards()[0].tensor.fill_(0)
+
+
+@pytest.mark.skipif(
+    not _TORCH_SHARDED_TENSOR_AVAILABLE, reason="Test requires the torch version to support `ShardedTensor`"
```

Comment:
lest add argument for `RunIf`

Reply:
I think we should put `@RunIf(min_torch="1.10.0")` because this tensor type will appear in 1.10

Reply:
@awaelchli I think if we put `@RunIf(min_torch="1.10.0")`, the test won't run for the nightly CI. Is that desirable?

Reply:
@yifuwang why so? I see in the CI we install  `1.10.0.dev20210816` and so if we put `min_torch=1.10.0` this condition should certainly hold up.

Reply:
@awaelchli I think dev releases come before the final release: https://www.python.org/dev/peps/pep-0440/#developmental-releases

```
>>> from packaging.version import Version
>>> Version("1.10.0.dev20210816") < Version("1.10.0")
True
```